### ISSUE TITLE
Support manipulating the origin

### ISSUE THREAD

**Issue by jussi-kalliokoski** [medialize/URI.js#210]
Matching the semantics of the `origin` in https://url.spec.whatwg.org/#concept-url-origin

e.g.

```javascript
URI("http://www.cdn.com/foo/bar").origin() // "http://www.cdn.com"
URI("http://www.cdn.com/foo/bar").origin("https://site.inter.net").toString() // "https://site.inter.net/foo/bar"
```

**Comment by rodneyrehm** [comment]
Please provide a proper description of what you want to achieve. It is important to *show* context, because terms usually have multiple meanings.

**Comment by jussi-kalliokoski** [comment]
@rodneyrehm:

> Please provide a proper description of what you want to achieve. It is important to show context, because terms usually have multiple meanings.

Sorry, messed up and the description got left out!

**Comment by rodneyrehm** [comment]
what's the expected difference of `.origin()` and [`.authority()`](http://medialize.github.io/URI.js/docs.html#accessors-authority) besides the scheme/protocol?

**Comment by justinmchase** [comment]
@rodneyrehm Probably nothing. What is the current way to get the authority and scheme from a url?

**Comment by rodneyrehm** [comment]
> Probably nothing. What is the current way to get the authority and scheme from a url?

There is no way to get scheme and authority in one go. Do you feel like adding `.origin()` is something you'd like to try and send a PR for?

**Comment by justinmchase** [comment]
Ok :)

https://github.com/medialize/URI.js/pull/261

I didn't see a `contibutors.md` so I probably missed some things. This is the barebones:

- the addition of the origin function
- unit tests

I ran the test via the index.html file and they passed. If I'm missing anyting required let me know and I'll fix it up.

**Comment by rodneyrehm** [comment]
> I didn't see a contibutors.md

yeah, a couple of things need to be done in this repository… I was hoping to do that with the 2.0 rewrite - whenever that may happen…

**Comment by rodneyrehm** [comment]
fixed by @justinmchase in #263

**Comment by rodneyrehm** [comment]
released in [v1.17.0](https://github.com/medialize/URI.js/releases/tag/v1.17.0)
